.github/workflows/publish.yml

```diff
@@ -1,29 +1,79 @@
 on:
   push:
     branches: main
+  pull_request:
+    branches: main
 
 name: Render and Publish
 
 jobs:
   build-deploy:
-    runs-on: ubuntu-latest
+    runs-on: self-hosted
     steps:
       - name: Check out repository
         uses: actions/checkout@v4
 
       - name: Set up Quarto
         uses: quarto-dev/quarto-actions/setup@v2
+        
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          use-public-rspm: true
+      - uses: r-lib/actions/setup-r-dependencies@v2
+
+      - name: Render Quarto Project
+        uses: quarto-dev/quarto-actions/render@v2
+
+      - name: Restore lychee cache
+        uses: actions/cache@v3
+        with:
+          path: .lycheecache
+          key: cache-lychee-${{ github.sha }}
+          restore-keys: cache-lychee-
+
+      - name: Link Checker
+        id: lychee
+        uses: lycheeverse/lychee-action@v1
         with:
-          # To install LaTeX to build PDF book
-          tinytex: true
-          # uncomment below and fill to pin a version
-          # version: SPECIFIC-QUARTO-VERSION-HERE
+          fail: true
+          args: "docs --exclude-loopback --insecure --exclude-mail --exclude-path docs/site_libs --cache --max-cache-age 1d"
 
-      # add software dependencies here
+      - name: Read lychee's out.md
+        if: failure()
+        id: lychee-output
+        uses: juliangruber/read-file-action@v1
+        with:
+          path: ./lychee/out.md
 
-      - name: Publish to GitHub Pages (and render)
+      - uses: mshick/add-pr-comment@v2
+        if: failure()
+        with:
+          message: ${{ steps.lychee-output.outputs.content }}
+
+      - name: Save lychee cache
+        uses: actions/cache/save@v3
+        if: always()
+        with:
+          path: .lycheecache
+          key: ${{ steps.restore-cache.outputs.cache-primary-key }}
+          
+      - name: Deploy gh-pages Preview
+        if: ${{ github.event_name == 'pull_request' && !contains(github.event.pull_request.title, 'chore(deps):') }}
         uses: quarto-dev/quarto-actions/publish@v2
```

Comment:
maybe that runs render a second time though :thinking:

Reply:
Yes, but the normal render before is needed for the link checker to do its work.

Reply:
Right! What a pity though, should we switch to something more homemade for the publication to GitHub Pages then? Like https://github.com/ropensci/dev_guide/blob/1aeade822cc26ef7fa7a235eadfc358be61f8279/.github/workflows/release.yml#L41-L53